code/_onclick/autoclick.dm

```diff
@@ -27,8 +44,31 @@
 	if(h)
 		. = h.CanItemAutoclick(object, location, params)
 
+/mob/proc/canMobMousedown(object, location, params)
+
+/mob/living/carbon/canMobMousedown(atom/object, location, params)
+	var/obj/item/H = get_active_held_item()
+	if(H)
+		. = H.canItemMouseDown(object, location, params)
+
 /obj/item/proc/CanItemAutoclick(object, location, params)
 
+/obj/item/proc/canItemMouseDown(object, location, params)
+	if(canMouseDown)
+		return src
+
+/obj/item/proc/onMouseDown(object, location, params, mob)
+	return
+
+/obj/item/proc/onMouseUp(object, location, params, mob)
+	return
+
+/obj/item/proc/onMouseDrag(src_object, over_object, src_location, over_location, params, mob)
```

Comment:
Specify the type of src_object, over_object, src_location, and over_location. Rename mob to mob/M